src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/Huffman.cs

```diff
@@ -10,272 +10,534 @@ namespace System.Net.Http.HPack
+            6,
+            5,
+            6,
+            5,
+            6,
+            6,
+            6,
+            5,
+            7,
+            7,
+            6,
+            6,
+            6,
+            5,
+            6,
+            7,
+            6,
+            5,
+            5,
+            6,
+            7,
+            7,
+            7,
+            7,
+            7,
+            15,
+            11,
+            14,
+            13,
+            28,
+            20,
+            22,
+            20,
+            20,
+            22,
+            22,
+            22,
+            23,
+            22,
+            23,
+            23,
+            23,
+            23,
+            23,
+            24,
+            23,
+            24,
+            24,
+            22,
+            23,
+            24,
+            23,
+            23,
+            23,
+            23,
+            21,
+            22,
+            23,
+            22,
+            23,
+            23,
+            24,
+            22,
+            21,
+            20,
+            22,
+            22,
+            23,
+            23,
+            21,
+            23,
+            22,
+            22,
+            24,
+            21,
+            22,
+            23,
+            23,
+            21,
+            21,
+            22,
+            21,
+            23,
+            22,
+            23,
+            23,
+            20,
+            22,
+            22,
+            22,
+            23,
+            22,
+            22,
+            23,
+            26,
+            26,
+            20,
+            19,
+            22,
+            23,
+            22,
+            25,
+            26,
+            26,
+            26,
+            27,
+            27,
+            26,
+            24,
+            25,
+            19,
+            21,
+            26,
+            27,
+            27,
+            26,
+            27,
+            24,
+            21,
+            21,
+            26,
+            26,
+            28,
+            27,
+            27,
+            27,
+            20,
+            24,
+            20,
+            21,
+            22,
+            21,
+            21,
+            23,
+            22,
+            22,
+            25,
+            25,
+            24,
+            24,
+            26,
+            23,
+            26,
+            27,
+            26,
+            26,
+            27,
+            27,
+            27,
+            27,
+            27,
+            28,
+            27,
+            27,
+            27,
+            27,
+            27,
+            26,
+            30
         };
 
         private static readonly ushort[] s_decodingTree = GenerateDecodingLookupTree();
 
         public static (uint encoded, int bitLength) Encode(int data)
         {
-            return _encodingTable[data];
+            return (s_encodingTableCodes[data], EncodingTableBitLengths[data]);
```

Comment:
Where is this used now?

Reply:
In unit tests only. 
I did some benchmarking in past with this result:
```C#
        [Benchmark]
        [Arguments(0)]
        public (uint, byte) ByTwoArrays(int i)
        {
            return (intArray1[i], byteArray2[i]);
        }
        [Benchmark]
        [Arguments(0)]
        public (uint, byte) ByTuppleArray(int i)
        {
            return tupleArray[i];
        }
```

|        Method | i |      Mean |     Error |    StdDev | Code Size |
|-------------- |-- |----------:|----------:|----------:|----------:|
|   ByTwoArrays | 0 | 4.3952 ns | 0.0771 ns | 0.0644 ns |      88 B |
| ByTuppleArray | 0 | 0.2959 ns | 0.0090 ns | 0.0070 ns |      41 B |

However, since it is not used anywhere but unit tests, there is no need to optimize it now.

Reply:
> I did some benchmarking in past with this result:

Thanks, but that's not a great comparison.  The ByTupleArray method is just returning the value found in an element in an array, whereas the ByTwoArrays method is taking the two values looked up, constructing a tuple struct from that, and returning that.  In something like the actual product loop, you need the two values individually, so you would actually need to be measuring the cost of extracting the two values from the tuple, and you wouldn't be paying to construct that tuple.

Try this instead:
```C#
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Diagnosers;
using BenchmarkDotNet.Running;
using System.Runtime.CompilerServices;

[MemoryDiagnoser]
public class Program
{
    static void Main(string[] args) => BenchmarkSwitcher.FromAssemblies(new[] { typeof(Program).Assembly }).Run(args);

    private (uint, byte)[] tupleArray = new[] { ((uint)1, (byte)1), ((uint)2, (byte)2), ((uint)3, (byte)3) };

    private uint[] intArray1 = new uint[] { 1, 2, 3 };

    private byte[] byteArray2 = new byte[] { 1, 2, 3 };

    [Benchmark]
    [Arguments(0)]
    public void ByTwoArrays(int i)
    {
        Use(intArray1[i], byteArray2[i]);
    }

    [Benchmark]
    [Arguments(0)]
    public void ByTupleArray(int i)
    {
        (uint ui, byte b) = tupleArray[i];
        Use(ui, b);
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    private static void Use(uint i, byte b) { }
}
```

I get results like:

|           Method | i |      Mean |     Error |    StdDev |
|----------------- |-- |----------:|----------:|----------:|
|      ByTwoArrays | 0 | 0.6738 ns | 0.0049 ns | 0.0044 ns |
|     ByTupleArray | 0 | 0.6740 ns | 0.0028 ns | 0.0022 ns |